src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Sse41.PlatformNotSupported.cs

```diff
@@ -182,12 +182,12 @@ public static class Sse41
         /// int _mm_extract_epi8 (__m128i a, const int imm8)
         ///   PEXTRB reg/m8, xmm, imm8
         /// </summary>
-        public static sbyte Extract(Vector128<sbyte> value, byte index) { throw new PlatformNotSupportedException(); }
+        public static int Extract(Vector128<sbyte> value, byte index) { throw new PlatformNotSupportedException(); }
```

Comment:
What impact does this have on containment, since it extracts to `reg/m8` and zero extends the upper bits...

Reply:
Not sure. There are some other similar instructions (e.g., PMOVZX* and PMOVSX* ), I will investigate later.

Reply:
What's the advantage of returning `int` instead of `sbyte`? 

The `int` is going to be zero extended byte that is pretty non-intuitive.

Reply:
@jkotas, the behavior of the `PEXTRB` instruction is to zero extend:
> Extract a byte integer value from xmm2 at the source byte offset specified by imm8 into reg or m8. The upper bits of r32 or r64 are zeroed.`

Reply:
Also, does the manual implementation zero extends the same way? I think the current manual implementation going to sign-extend, not zero-extended.

Reply:
> the behavior of the PEXTRB instruction is to zero extend

Right. And it makes the behavior very non-intuitive when it returns int. I think this instruction should be either defined for `Vector128<byte>` only; or it should return `sbyte`.

Reply:
The manual implementation will probably sign-extend exactly because the return type was `sbyte` instead of `int`. Thing is, the instruction extracts a byte into a 32 bit register, it doesn't exactly deal with signedness. To get `sbyte` to work properly you'd need to add another instruction - `movsx eax, al` - that's a bit unusual for an intrinsic.

Reply:
@jkotas, it was being changed from `sbyte` to `int` due to `https://github.com/dotnet/coreclr/issues/17957`.

However, I agree that just exposing `byte` might make more sense. It might also reduce confusion for the case where we write directly to memory, and zero extension cannot happen.

Reply:
@jkotas Thanks for capturing this. 

I also agree to just remove the `sbyte` and `short` overloads rather than changing the return type to `int/uint`.